docs/OneBlinkFormControlled.md

```diff
@@ -0,0 +1,230 @@
+        {},
+        newFormSubmission,
+        {
+          formsAppId,
+          jobId: null,
+          externalId: null,
+          draftId: null,
+          preFillFormDataId: null,
+        },
+      )
+
+      setSubmitState({
+        formSubmissionResult: null,
+        submitError: null,
+        isSubmitting: true,
+      })
+
+      try {
+        const newFormSubmissionResult = await submissionService.submit({
+          formSubmission,
+        })
+        if (
+          newFormSubmissionResult.isOffline &&
+          !newFormSubmissionResult.isInPendingQueue
+        ) {
+          throw new OneBlinkAppsError(
+            'You cannot submit this form while offline, please try again when connectivity is restored.',
+            {
+              isOffline: true,
+            },
+          )
+        }
+
+        if (isMounted.current) {
+          setSubmitState({
+            formSubmissionResult: newFormSubmissionResult,
+            isSubmitting: false,
+            submitError: null,
+          })
+        }
+      } catch (error) {
+        if (isMounted.current) {
+          setSubmitState({
+            formSubmissionResult: null,
+            isSubmitting: false,
+            submitError: error,
+          })
+        }
+      }
+    },
+    [isMounted],
+  )
+
+  const handleSaveDraft = React.useCallback(
+    async (newDraftSubmission: FormTypes.NewDraftSubmission) => {
+      const draftSubmission: FormTypes.DraftSubmission = {
+        ...newDraftSubmission,
+        formsAppId,
+      }
+      setSaveDraftState({
+        saveDraftError: null,
+        isSavingDraft: true,
+      })
+
+      try {
+        await draftService.addDraft(
+          {
+            title: form.name,
+            formId: form.id,
+            externalId: null,
+            jobId: null,
+          },
+          draftSubmission,
+        )
+
+        if (isMounted.current) {
+          setSaveDraftState({
+            saveDraftError: null,
+            isSavingDraft: false,
+          })
+        }
+      } catch (error) {
+        if (isMounted.current) {
+          setSaveDraftState({
+            saveDraftError: error,
+            isSavingDraft: false,
+          })
+        }
+      }
+    },
+    [isMounted],
+  )
+
+  const handleCancel = React.useCallback(() => {
+    // handle cancel here...
+  }, [])
+
+  if (isSubmitting) {
+    // Render submitting animation/loading
+  }
+
+  if (submitError) {
+    // Render error while submitting
+  }
+
+  if (isSavingDraft) {
+    // Render saving draft animation/loading
+  }
+
+  if (saveDraftError) {
+    // Render error while saving draft
+  }
+
+  if (formSubmissionResult) {
+    // Render submission success
+  }
+
+  return (
+    <OneBlinkFormControlled
+      captchaSiteKey={captchaSiteKey}
+      googleMapsApiKey={googleMapsApiKey}
+      formsAppId={formsAppId}
+      definition={definition}
+      submission={submission}
+      setFormSubmission={setFormSubmission}
```

Comment:
Should this just be `onSubmissionChange` ?

Reply:
I wanted it to be clear that this is not just a callback type prop, but a react state setter.